tests/test_cloud.py

```diff
@@ -1740,6 +1748,596 @@ def test_extract_tar(self, container_client_mock, compression, file_ext, tmpdir)
+                # Create object and call list_bucket
+                cloud_interface = GoogleCloudInterface(
+                    "https://console.cloud.google.com/storage/browser/barman-tests/path/to/somewhere"
+                )
+                content = cloud_interface.list_bucket(
+                    test_case["prefix"], delimiter=delimiter
+                )
+                assert content == test_case["expected"]
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 5), reason="requires python3.5 or higher"
+    )
+    def test_upload_fileobj_with(self):
+        """
+        Tests the tags argument is provided to the container client when uploading
+        a file if tags are provided when creating AzureCloudInterface.
+        """
+        test_cases = {
+            "No tag": {
+                "cloud_interface_tags": None,
+                "override_tags": None,
+                "expected_tags": None,
+            },
+            "Cloud interface tags are used if no override tags": {
+                "cloud_interface_tags": [("foo", "bar"), ("baz $%", "qux -/")],
+                "override_tags": None,
+                "expected_tags": {"foo": "bar", "baz $%": "qux -/"},
+            },
+            "Override tags are used in place of cloud interface tags": {
+                "cloud_interface_tags": [("foo", "bar")],
+                "override_tags": [("$+ a", "///"), ("()", "[]")],
+                "expected_tags": {"$+ a": "///", "()": "[]"},
+            },
+        }
+        for test_name, test in test_cases.items():
+            with self.subTest(name=test_name):
+                with unittest_patch(
+                    "barman.cloud_providers.google_cloud_storage.storage.Client"
+                ) as gcs_client_mock:
+                    mock_fileobj = mock.MagicMock()
+                    mock_blob = mock.MagicMock()
+
+                    service_client_mock = gcs_client_mock.return_value
+                    container_client_mock = service_client_mock.bucket.return_value
+                    container_client_mock.blob.return_value = mock_blob
+                    # Init metadata to none for no tag case
+                    mock_blob.metadata = None
+                    cloud_interface = GoogleCloudInterface(
+                        "https://console.cloud.google.com/storage/browser/barman-test/test/path/to/my/",
+                        tags=test["cloud_interface_tags"],
+                    )
+                    mock_key = "path/to/blob"
+                    cloud_interface.upload_fileobj(
+                        mock_fileobj, mock_key, override_tags=test["override_tags"]
+                    )
+                    # Validate behavior
+                    assert mock_blob.metadata == test["expected_tags"]
+                    container_client_mock.blob.assert_called_once_with(mock_key)
+                    mock_blob.upload_from_file.assert_called_once_with(mock_fileobj)
+
+    # @mock.patch.dict(
```

Comment:
Yes I left it on purpose for code review and planed to get rid of remaining commented code after. I though it would be easier to get differences wit other providers this way, but maybe not.